source/grid/grid_tools.cc

```diff
@@ -2279,7 +2279,7 @@ namespace GridTools
         }
 
     // Creating a Bounding Box for all cells on the chosen refinement_level
-    for (typename MeshType::cell_iterator cell: mesh.cell_iterators_on_level(refinement_level))
+    for (const typename MeshType::cell_iterator &cell: mesh.cell_iterators_on_level(refinement_level))
```

Comment:
Since these are iterators I usually omit the reference here. I am not sure if this makes any significant difference.

Reply:
It probably doesn't make much of a difference. An iterator consists of two integers and two pointers. If you take it by reference, the compiler will just propagate the reference to where it is used and take the structure apart there. If you take it by value, the compiler will scalarize the structure  into four independent variables and propagate these independently. I would be surprised if it makes any kind of measurable difference.